api/base/serializers.py

```diff
@@ -177,8 +234,27 @@ def to_representation(self, obj, envelope='data'):
         meta = getattr(self, 'Meta', None)
         type_ = getattr(meta, 'type_', None)
         assert type_ is not None, 'Must define Meta.type_'
-        data = super(JSONAPISerializer, self).to_representation(obj)
-        data['type'] = type_
+
+        data = collections.OrderedDict([('id', ''), ('type', type_), ('attributes', collections.OrderedDict()),
+                                        ('relationships', collections.OrderedDict()), ('links', {})])
+
+        fields = [field for field in self.fields.values() if not field.write_only]
+
+        for field in fields:
+            try:
+                attribute = field.get_attribute(obj)
+            except SkipField:
+                continue
+
+            if isinstance(field, HyperlinkedIdentityFieldWithMeta):
```

Comment:
A`HyperlinkedRelatedField` is looking for a `pk` (or _id).   This allows it to resolve into a list of detail view url's. From the docs(http://www.django-rest-framework.org/api-guide/relations/#hyperlinkedrelatedfield)
![screen shot 2015-09-01 at 11 34 34 pm](https://cloud.githubusercontent.com/assets/9755598/9622593/fa71aaae-5102-11e5-9b54-620285947565.png)
 We wanted to return the url for the list view of contributors, say, not a list of detail-views of contributors.
![screen shot 2015-09-02 at 12 15 31 am](https://cloud.githubusercontent.com/assets/9755598/9622916/6f8ca498-5107-11e5-8ace-bf5f1257b26f.png)

Here's what HyperlinkedRelatedField would resolve to for node contributors:
![screen shot 2015-09-02 at 12 03 29 am](https://cloud.githubusercontent.com/assets/9755598/9622984/a1aa97ae-5108-11e5-8428-b168130e4896.png)
![screen shot 2015-09-02 at 12 15 12 am](https://cloud.githubusercontent.com/assets/9755598/9622915/69ffed1e-5107-11e5-812d-b645719e0a08.png)